draft-yasskin-http-origin-signed-responses.md

```diff
@@ -1054,112 +1054,76 @@ This allows the mitigation against downgrades in {{seccons-downgrades}}, but
 prohibits intermediates from providing a cache of the validity information. We
 could do both with a list of URLs.
 
-## application/http-exchange+cbor format for HTTP/1 compatibility # {#application-http-exchange}
+## application/signed-exchange format # {#application-signed-exchange}
+
+To allow signed exchanges to be the targets of `<link rel=prefetch>` tags, we
+define the  `application/signed-exchange` content type that represents a signed
+HTTP exchange, including request metadata and header fields, response metadata
+and header fields, and a response payload.
+
+This content type consists of the concatenation of the following items:
+
+1. The ASCII characters "sxg1" followed by a 0 byte, to serve as a file
+   signature.
+
+   Note: RFC EDITOR PLEASE DELETE THIS NOTE; The implementation of the final RFC
+   MUST use this file signature, but implementations of drafts MUST NOT use it
+   and MUST use another implementation-specific string beginning with "sxg1-" and
+   ending with a 0 byte instead.
```

Comment:
I might suggest instead saying that draft impls not use the final MIME type instead/additionally. Consider if this format changes a bit. Probably someone will end up in a situation where they need to support two iterations at the same time. That means you first need to read enough to dispatch on the type (which isn't even fixed-width), then parse the rest out. Or you cleverly meld the state machines.

Whereas, if it's in the content type, the HTTP stack has already given you the headers as one atomic thing, so you can go ahead and dispatch there.

Edit: Ah, I see you've put the parameter in the MIME type registration. Maybe make it clearer here that that's going on?

Reply:
I think the version identifier has to go in both because people will store these as files, and their servers need to be able to pick the right mime type. But yeah, I've now said the client needs to trust the mime type.